File: Application/src/com/hhp227/application/SplashActivity.java
package com.hhp227.application;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.view.Window;

public class SplashActivity extends Activity {

	// 뜨는 시간
	private static int SPLASH_TIME_OUT = 1250;

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		// 액션바 안보이기
		this.requestWindowFeature(Window.FEATURE_NO_TITLE);
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_splash);

		new Handler().postDelayed(new Runnable() {

			@Override
			public void run() {
				// 화면이 닫히고 전환됨
				Intent intent = new Intent(SplashActivity.this, MainActivity.class);
				startActivity(intent);

				finish();
			}
		}, SPLASH_TIME_OUT);
	}

}

File: app/src/main/java/com/hhp227/application/activity/SplashActivity.kt
package com.hhp227.application.activity

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.flowWithLifecycle
import androidx.lifecycle.lifecycleScope
import com.hhp227.application.R
import com.hhp227.application.app.AppController
import com.hhp227.application.databinding.ActivitySplashBinding
import com.hhp227.application.util.InjectorUtils
import com.hhp227.application.viewmodel.SplashViewModel
import kotlinx.coroutines.delay
import kotlinx.coroutines.flow.launchIn
import kotlinx.coroutines.flow.onEach

class SplashActivity : AppCompatActivity() {
    private val viewModel: SplashViewModel by viewModels {
        InjectorUtils.provideSplashViewModelFactory()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(ActivitySplashBinding.inflate(layoutInflater).root)

        viewModel.userFlow.flowWithLifecycle(lifecycle, Lifecycle.State.STARTED).onEach { user ->
            val intent = Intent(this, user?.let { MainActivity::class.java } ?: LoginActivity::class.java)

            delay(SPLASH_TIME_OUT)
            startActivity(intent)
            finish()
            overridePendingTransition(R.anim.splash_in, R.anim.splash_out)
        }.launchIn(lifecycleScope)
    }

    companion object {
        private const val SPLASH_TIME_OUT = 1250L
    }
}